packages/components/nodes/memory/MementoMemory/MementoMemory.ts

```diff
@@ -0,0 +1,77 @@
+import { SystemChatMessage } from 'langchain/schema'
+import { INode, INodeData, INodeParams } from '../../../src/Interface'
+import { getBaseClasses } from '../../../src/utils'
+import { MomentoChatMessageHistory } from 'langchain/stores/message/momento'
+import { ICommonObject } from '../../../src'
+import { CacheClient } from '@gomomento/sdk'
+
+class MomentoMemory implements INode {
+    label: string
+    name: string
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'Momento Memory'
+        this.name = 'MomentoMemory'
+        this.type = 'MomentoMemory'
+        this.icon = 'memory.svg'
+        this.category = 'Memory'
+        this.description = 'Summarizes the conversation and stores the memory in Momento server'
+        this.baseClasses = [this.type, ...getBaseClasses(MomentoChatMessageHistory)]
+        this.inputs = [
+            {
+                label: 'Cache Name',
+                name: 'cacheName',
+                type: 'string',
+                default: 'langchain'
+            },
+            {
+                label: 'Session TTL',
+                name: 'sessionTtl',
+                type: 'number',
+                description: 'The time-to-live for the session in seconds',
+                default: 300,
+                optional: true
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
+        const cacheName = nodeData.inputs?.cacheName as string
+        const sessionTtl = nodeData.inputs?.sessionTtl as number
+
+        const sessionId = options?.chatId as string
+        const client = options?.client as CacheClient
```

Comment:
how does the `client` get passed into the `options`?

Reply:
we have to initialize a new CacheClient object dont we? - https://js.langchain.com/docs/modules/memory/examples/momento